src/agents/workflows/subgraphs/step1_preprocessing/preprocessing_utils.py

```
@@ -17,6 +19,48 @@
+    if not filename:
+        filename = "downloaded_file"
+
+    local_path = target_dir / filename
+
+    logger.info(f"Downloading from URL: {url[:100]}... -> {local_path}")
+
+    try:
+        # User-Agent 헤더 추가 (일부 서버에서 필요)
+        req = urllib.request.Request(
+            url, headers={"User-Agent": "Mozilla/5.0 (Proovy-AI/1.0)"}
+        )
```

Ruff (0.15.0)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)